### ISSUE TITLE
Can't get InvertibleScrollView to work

### ISSUE THREAD

**Issue by satya164** [650Industries/react-native-scrollable-mixin#1]
I'm trying to use this on react native for Android. But the it renders the same as the regular list view.

My React Native version is **0.11.0**

Here is my code,

``` js
import React from "react-native";
import InvertibleScrollView from "react-native-invertible-scroll-view";
import ChatItem from "./chat-item";
import PageLoading from "./page-loading";
import PageRetry from "./page-retry";

const {
    ListView,
    View
} = React;

export default class Chat extends React.Component {
    constructor(props) {
        super(props);

        let ds = new ListView.DataSource({ rowHasChanged: (r1, r2) => r1 !== r2 });

        this.state = {
            dataSource: ds.cloneWithRows(this.props.data)
        };
    }

    componentWillReceiveProps(nextProps) {
        this.setState({
            dataSource: this.state.dataSource.cloneWithRows(nextProps.data)
        });
    }

    componentDidMount() {
        if (this._scroll) {
            this._scroll.scrollTo(0);
        }
    }

    render() {
        return (
            <View {...this.props}>
                {(() => {
                    if (this.props.data.length) {
                        return (
                            <ListView
                                renderScrollComponent={props => <InvertibleScrollView {...props} inverted ref={c => this._scroll = c} />}
                                dataSource={this.state.dataSource}
                                renderRow={(text, sectionID, rowID) => {
                                    let previousText;

                                    if (rowID > 0) {
                                        previousText = this.state.dataSource.getRowData(0, rowID - 1);
                                    }

                                    return <ChatItem key={text.id} text={text} previousText={previousText} />;
                                }}
                            />
                        );
                    } else if (this.props.failed) {
                        return <PageRetry onRetry={this._onRetry.bind(this)} />;
                    } else {
                        return <PageLoading />;
                    }
                })()}
            </View>
        );
    }
}

Chat.propTypes = {
    failed: React.PropTypes.bool,
    data: React.PropTypes.arrayOf(React.PropTypes.shape({
        id: React.PropTypes.string.isRequired
    })).isRequired
};
```

Am I doing something wrong?

**Comment by ide** [comment]
I haven't tried this on Android. Matrix transforms might not be supported yet.

**Comment by satya164** [comment]
@ide Just noticed that I filed the issue in the wrong repo. Sorry :(

Will it work if I use `scaleY(-1)` instead of matrix transformations? I kinda need it very badly :(

**Comment by ide** [comment]
You could try that and report back if it works. I believe that `scaleY` is just converted to a matrix transformation though.

**Comment by satya164** [comment]
Will do. Thanks.

**Comment by satya164** [comment]
@ide It seems to be working, I just change the following code,

```
let styles = StyleSheet.create({
  verticallyInverted: {
    transformMatrix: [
       1,  0,  0,  0,
       0, -1,  0,  0,
       0,  0,  1,  0,
       0,  0,  0,  1,
    ],
  },
  horizontallyInverted: {
    transformMatrix: [
      -1,  0,  0,  0,
       0,  1,  0,  0,
       0,  0,  1,  0,
       0,  0,  0,  1,
    ],
  },
});
```

to

```
let styles = StyleSheet.create({
  verticallyInverted: {
    transform: [
        { scaleY: -1 }
    ],
  },
  horizontallyInverted: {
    transform: [
        { scaleX: -1 }
    ],
  },
});
```

Shall I send a pull request?

Thanks a ton for this.

**Comment by ide** [comment]
Yes! Please do! (note that our code style uses two spaces for indentation)

**Comment by satya164** [comment]
@ide Okies. Will do :)

**Comment by ide** [comment]
Published react-native-invertible-scroll-view 0.1.2 to npm. Thank you for your contribution.

**Comment by satya164** [comment]
@ide Awesome. Thanks a lot.
